lib/active_merchant/billing/gateways/nuvei.rb

```diff
@@ -262,6 +310,14 @@ def message_from(response)
         reason = response[:reason]&.present? ? response[:reason] : nil
         response[:gwErrorReason] || reason || response[:transactionStatus]
       end
+
+      def build_payment_post(post, money, payment, options = {})
```

Comment:
💬 Comment

There is a possible refactor where you build the post objetc in the `authorize` method and the use that object on the remaining transaction.